raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -766,6 +827,31 @@ contract TokenNetwork is Utils {
         signature_address = ECVerify.ecverify(message_hash, non_closing_signature);
     }
 
+    function recoverAddressFromCooperativeSettleSignature(
+        uint256 channel_identifier,
+        address participant1,
+        uint256 participant1_balance,
+        address participant2,
```

Comment:
From chat: 

> some context about me adding the addresses in the signature (not only the balance values) is - only having the balance values there in an unenforced order, can make it confusing to determine/check whose is the balance.

Why would it make it confusing? First balance would be participant1's balance and second balance participant2 balance.

Reply:
Yes, but in the Raiden client side: - you need to choose an order first and then send it to the other participant to sign. This is were I see issues - you have to pay extra care that the node receiving this message to sign knows how to interpret the balances correctly. 
I think this format brings more clarity.

Reply:
> This is were I see issues - you have to pay extra care that the node receiving this message to sign knows how to interpret the balances correctly.

Each node will know its balance and the balance of its counterparty so I still don't see the scenario where having the addresses in the signature is really required.

Would like to hear @hackaugusto's opinion on this too.

> I think this format brings more clarity.

True, but this signature is intended to be made and consumed by a raiden client, so human readable clarity is not a goal here.

Reply:
I agree. Changes to the signature if needed can happen in another PR. Whenever @hackaugusto becomes available we can see what his opinion is.

Reply:
I'm fine with either. having the participant addresses is a bit more explicit wich is nice